packages/wrangler/src/__tests__/pages-deployment-tail.test.ts

```diff
@@ -0,0 +1,772 @@
+import MockWebSocket from "jest-websocket-mock";
+import { Headers, Request } from "undici";
+import { mockAccountId, mockApiToken } from "./helpers/mock-account-id";
+import { setMockResponse, unsetAllMocks } from "./helpers/mock-cfetch";
+import { mockConsoleMethods } from "./helpers/mock-console";
+import { useMockIsTTY } from "./helpers/mock-istty";
+import { msw, mswScriptHandlers } from "./helpers/msw";
+import { runInTempDir } from "./helpers/run-in-tmp";
+import { runWrangler } from "./helpers/run-wrangler";
+import type {
+	TailEventMessage,
+	RequestEvent,
+	ScheduledEvent,
+	AlarmEvent,
+} from "../tail/createTail";
+import type { RequestInit } from "undici";
+import type WebSocket from "ws";
+describe("pages deployment tail", () => {
+	beforeEach(() => msw.use(...mswScriptHandlers));
+	runInTempDir();
+	mockAccountId();
+	mockApiToken();
+
+	const std = mockConsoleMethods();
+
+	afterEach(() => {
+		mockWebSockets.forEach((ws) => ws.close());
+		mockWebSockets.splice(0);
+		unsetAllMocks();
+	});
+
+	/**
+	 * Interaction with the tailing API, including tail creation,
+	 * deletion, and connection.
+	 */
+	describe("API interaction", () => {
+		it("should throw an error if deployment isn't provided", async () => {
+			await expect(runWrangler("pages deployment tail")).rejects.toThrow();
+		});
+
+		it("creates and then delete tails by deployment ID", async () => {
+			const api = mockTailAPIs();
+			expect(api.requests.creation.length).toStrictEqual(0);
+
+			await runWrangler(
+				"pages deployment tail mock-deployment-id --project-name mock-project"
+			);
+
+			await expect(api.ws.connected).resolves.toBeTruthy();
+			expect(api.requests.creation.length).toStrictEqual(1);
+			expect(api.requests.deletion.count).toStrictEqual(0);
+
+			api.ws.close();
+			expect(api.requests.deletion.count).toStrictEqual(1);
+		});
+
+		it("creates and then delete tails by project name", async () => {
+			const api = mockTailAPIs();
+			expect(api.requests.creation.length).toStrictEqual(0);
+
+			await runWrangler("pages deployment tail --project-name mock-project");
+
+			await expect(api.ws.connected).resolves.toBeTruthy();
+			expect(api.requests.creation.length).toStrictEqual(1);
+			expect(api.requests.deletion.count).toStrictEqual(0);
+
+			api.ws.close();
+			expect(api.requests.deletion.count).toStrictEqual(1);
+		});
+
+		it("errors when the websocket closes unexpectedly", async () => {
+			const api = mockTailAPIs();
+			api.ws.close();
+
+			await expect(
+				runWrangler(
+					"pages deployment tail mock-deployment-id --project-name mock-project"
+				)
+			).rejects.toThrow();
```

Comment:
does this throw a specific error? if so we could test that as well

Reply:
Yeeeee I could specify some command output

Reply:
Added output snapshot